pkg/build/builder/util.go

```diff
@@ -24,5 +24,11 @@ func getBuildEnvVars(build *buildapi.Build) map[string]string {
 		build.Parameters.Revision.Git.Commit != "" {
 		envVars["OPENSHIFT_BUILD_COMMIT"] = build.Parameters.Revision.Git.Commit
 	}
+	if build.Parameters.Strategy.Type == buildapi.STIBuildStrategyType {
```

Comment:
this will allow users to override the OPENSHIFT_BUILD* env vars which might break the build, right? Any reason not to use the withoutDuplicate merging function here?